substrate/frame/contracts/src/lib.rs

```diff
@@ -214,6 +214,18 @@ pub struct Environment<T: Config> {
 	block_number: EnvironmentType<BlockNumberFor<T>>,
 }
 
+/// Defines the current version of the HostFn APIs.
+/// This is used to communicate the available APIs in pallet-contracts.
+///
+/// The version is bumped any time a new HostFn is added or stabilized.
```

Comment:
I'm thinking about if we're better off using more general semantics here. Something like `impl_version` that we generally increment every time a host fn is added/stabilized, but also every time we want to communicate some other change, like an important bug fix.

Reply:
I think this will be difficult to do with a linear numbers. Bugfixes will get back ported and then this system will fall apart.